weldx/measurement.py

```diff
@@ -37,71 +28,705 @@ class Signal:
 
     signal_type: str
     unit: str
-    data: Union[Data, None]
+    data: "TimeSeries" = None
+
+    def __post_init__(self):
+        """Perform some checks after construction."""
+        if self.signal_type not in ["analog", "digital"]:
+            raise ValueError(f"{self.signal_type} is an invalid signal type.")
 
 
 @dataclass
-class DataTransformation:
-    """Simple dataclass implementation for signal transformations."""
+class SignalTransformation:
+    """Describes the transformation of a signal."""
 
     name: str
-    input_signal: Signal
-    output_signal: Signal
     error: Error
-    func: str = None
-    meta: str = None
+    func: "MathematicalExpression" = None
+    type_transformation: str = None
+    input_shape: Tuple = None
+    output_shape: Tuple = None
+
+    def __post_init__(self):
+        """Perform some tests after construction."""
+        if self.type_transformation is not None:
+            self.type_transformation = self.type_transformation.upper()
+            if self.type_transformation not in ["AA", "AD", "DA", "DD"]:
+                raise ValueError(
+                    f"Invalid type transformation: {self.type_transformation}\n"
+                    "Valid values are 'AA', 'AD', 'DA' and 'DD'."
+                )
+
+        if self.func is None and self.type_transformation is None:
+            raise ValueError("No transformation specified")
+
+        if self.func is not None:
+            self._evaluate_function()
+
+    def _evaluate_function(self):
+        """Evaluate the internal function."""
+        variables = self.func.get_variable_names()
+        if len(variables) != 1:
+            raise ValueError("The provided function must have exactly one parameter")
 
 
 @dataclass
-class Source:
+class SignalSource:
     """Simple dataclass implementation for signal sources."""
 
     name: str
     output_signal: Signal
     error: Error
 
 
+# equipment ----------------------------------------------------------------------------
 @dataclass
-class MeasurementChain:
-    """Simple dataclass implementation for measurement chains."""
+class GenericEquipment:
```

Comment:
Maybe we should rename this to `MeasurementEquipment` since it is rather specific to this application ?
My initial idea behind `GenericEquipment` was to provide some generic metadata that can be used for all kinds of equipment (robots, power sources, sensors etc.)

So other equipment-classes like `MeasurementEquipment` could inherit from that class

Reply:
Should all the files (tag, schemas) be moved to the measurement directory then?